runtime/client/queue.go

```diff
@@ -65,7 +75,93 @@ func (m *MockQueue) CreateArtifact(taskID, runID, name string, payload *queue.Po
 	return args.Get(0).(*queue.PostArtifactResponse), args.Error(1)
 }
 
-var AnyPostArtifactRequest = mock.MatchedBy(func(i interface{}) bool {
+var PostAnyArtifactRequest = mock.MatchedBy(func(i interface{}) bool {
 	_, ok := i.(*queue.PostArtifactRequest)
 	return ok
 })
+
+var PostS3ArtifactRequest = mock.MatchedBy(func(i interface{}) bool {
+	r, ok := i.(*queue.PostArtifactRequest)
+	if !ok {
+		return false
+	}
+	var s3req queue.S3ArtifactRequest
+	if json.Unmarshal(*r, &s3req) != nil {
+		return false
+	}
+	return s3req.StorageType == "s3"
+})
+
+// ExpectS3Artifact will setup queue to expect an S3 artifact with given
+// name to be created for taskID and runID using m and returns
+// a channel which will receive the artifact.
+func (m *MockQueue) ExpectS3Artifact(taskID string, runID int, name string) <-chan []byte {
```

Comment:
This method is super nice for when we have to test that artifacts are created :)